src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -187,6 +187,14 @@ async Task TrackResourceSnapshotsAsync()
                 {
                     SetSelectedResourceOption(selectedResource);
                 }
+                else
+                {
+                    var resourcesWithDisplayName = _resourceByName.Values.Where(r => string.Equals(ResourceName, r.DisplayName, StringComparisons.ResourceName)).ToList();
+                    if (resourcesWithDisplayName.Count == 1)
+                    {
+                        SetSelectedResourceOption(resourcesWithDisplayName.Single());
+                    }
```

Comment:
Add a helper method for getting the resource from the dictionary. It first checks the exact name, then falls back to getting it via display name. Then you can remove this else.